improvement/ZENKO-1753 split tests into parallel stages

ssalaues opened the pull request:
**What does this PR do, and why do we need it?**
- Splits the tests into parallel stages for quicker build times
- Slight rearrangement of the make steps to wait less for stabilization
- Bypasses pre-merge tests if on a `documentation` branch that only contains changes within the docs folder
- Removes unnecessary ci-values putting the zenko installs closer to a default

**Which issue does this PR fix?**
ZENKO-1753

**Special notes for your reviewers**:

Commit:
ci: split tests into parallel stages

bert-e commented:
# Hello ssalaues,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-1753 contains:


* `Zenko Backlog`


Considering where you are trying to merge, I expected to find:


* `1.0.3`

* `1.1.0`


Please check the `Fix Version/s` of ZENKO-1753, or the target
branch of this pull request.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/1.1/improvement/ZENKO-1753-ci-stages` with contents from `improvement/ZENKO-1753-ci-stages`
and `development/1.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/1.1/improvement/ZENKO-1753-ci-stages origin/development/1.1
 $ git merge origin/improvement/ZENKO-1753-ci-stages
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/1.1/improvement/ZENKO-1753-ci-stages
```

Commit:
improvement: remove unnecessary ci values

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/1.1/improvement/ZENKO-1753-ci-stages` with contents from `improvement/ZENKO-1753-ci-stages`
and `development/1.1`.

Please resolve the conflict on **the integration branch** (`w/1.1/improvement/ZENKO-1753-ci-stages`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/1.1/improvement/ZENKO-1753-ci-stages
 $ git pull  # or "git reset --hard origin/w/1.1/improvement/ZENKO-1753-ci-stages"
 $ git merge origin/development/1.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/ZENKO-1753-ci-stages
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/1.1/improvement/ZENKO-1753-ci-stages
```

Commit:
ci: do not run e2e suite on docs only branches

This commit adds a script to check for a 'documentation/*' branch. If
no files outside of the docs folder are changed then the e2e tests are
skipped. If the diff includes files modified outside of the docs folder
then the e2e is ran.

bert-e commented:
# History mismatch



Merge commit #4c943d18d3b57f2b407df2c079b24e61385cbf78 on the integration branch
`w/1.1/improvement/ZENKO-1753-ci-stages` is merging a branch which is neither the current
branch `improvement/ZENKO-1753-ci-stages` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/ZENKO-1753-ci-stages` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/ZENKO-1753-ci-stages` into
`development/1.0`
* [w/1.1/improvement/ZENKO-1753-ci-stages](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/ZENKO-1753-ci-stages) will be merged into `development/1.1` (pull request #654)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tmacro

* @nicolast

* @ssalaues

* @vrancurel

bert-e commented:
# History mismatch



Merge commit #11887264fbd03052b01adcc2bf8278d9ee789c27 on the integration branch
`w/1.1/improvement/ZENKO-1753-ci-stages` is merging a branch which is neither the current
branch `improvement/ZENKO-1753-ci-stages` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/ZENKO-1753-ci-stages` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #11887264fbd03052b01adcc2bf8278d9ee789c27 on the integration branch
`w/1.1/improvement/ZENKO-1753-ci-stages` is merging a branch which is neither the current
branch `improvement/ZENKO-1753-ci-stages` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/ZENKO-1753-ci-stages` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tmacro

* @nicolast

* @ssalaues

* @vrancurel

JianqinWang reviewed (approved):
🍾 🎉

bert-e commented:
# History mismatch



Merge commit #404446922ddc0da472dd50dc3c70d948a767b85e on the integration branch
`w/1.1/improvement/ZENKO-1753-ci-stages` is merging a branch which is neither the current
branch `improvement/ZENKO-1753-ci-stages` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/ZENKO-1753-ci-stages` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tmacro

* @nicolast

* @ssalaues

* @vrancurel

ssalaues commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`

* :heavy_check_mark: `development/1.1`



The following branches will **NOT** be impacted:

* `development/0.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`

* :heavy_check_mark: `development/1.1`



The following branches have **NOT** changed:

* `development/0.9`



Please check the status of the associated issue ZENKO-1753.

Goodbye ssalaues.
